xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererUtil.cpp

```diff
@@ -43,32 +44,31 @@ static uint32_t build_rgba(int yuv[3], int alpha, bool mergealpha)
 }
 #undef clamp
 
-std::vector<uint32_t> convert_rgba(CDVDOverlayImage* o, bool mergealpha)
+void convert_rgba(CDVDOverlayImage* o, bool mergealpha, std::vector<uint32_t>& rgba)
```

Comment:
This seems odd. Is there a reason why you want to pass in an empty vector by reference rather than allocate the vector in the method and return it?

Reply:
Isn't this a best practice to avoid copying the vector later when it already has elements? We seem do to this everywhere

Reply:
I haven't looked deeply, but RVO - return value optimization - might be a reason to allocate the vector in the method and return it

Reply:
Yes probably this is the reason. I want to be sure that a copy of the object is not produced. I know these are common optimizations in c++ but they are still compiler dependent and these RVO and NRVO optimizations are not always guaranteed to occur.

And from a cosmetic point of view seems not looks bad as is in line with other calls:

```c++
  convert_rgba(o, USE_PREMULTIPLIED_ALPHA, min_x, max_x, min_y, max_y, rgba);

  glGenTextures(1, &m_texture);
  glBindTexture(GL_TEXTURE_2D, m_texture);

  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP);
  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP);
  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);

  LoadTexture(GL_TEXTURE_2D, max_x - min_x, max_y - min_y, o->width * 4, &m_u, &m_v, false,
              rgba.data() + min_x + min_y * o->width);

  glBindTexture(GL_TEXTURE_2D, 0);
```

Reply:
As time go on, some compiler-specific optimizations do become guaranteed actually. I'm pretty sure that RVO is one of those and is now guaranteed